app/gui2/shared/languageServer.ts

```diff
@@ -152,8 +153,13 @@ export class LanguageServer extends ObservableV2<Notifications & TransportEvents
     this.client.onError((error) => {
       console.error('Unexpected Language Server connection error:', error)
     })
-    transport.on('error', (error) => console.error('Language Server transport error:', error))
+    transport.on('error', (error) => {
+      if (this.shouldReconnect) {
+        console.error('Language Server transport error:', error)
+      }
+    })
     const reinitializeCb = () => {
+      if (!this.shouldReconnect) return
```

Comment:
The reconnecting mechanism is inside websocket inside `transport` - we need to call `close` if we don't want to reconnect anymore.

Reply:
i guess the issue (i think) is that `onAbort` seems to run quite a bit later than `stopReconnecting`, so without this condition then it will try reconnecting after the final `'close'` (i *think*)

Reply:
note that `dispose()` calls `clientScope.dispose()` which in turn calls `transport.close()`

Reply:
> i guess the issue (i think) is that `onAbort` seems to run quite a bit later than `stopReconnecting`, so without this condition then it will try reconnecting after the final `'close'` (i _think_)

I mean, I'm not against this condition, but this line will prevent _initializing_, not reconnecting itself. The reconnecting happens in `transport`'s websocket, regardless of `shouldReconnect` flag here. Therefore, on `close` event, if `shouldReconnect` flag is true, we should manually close transport (which will close websocket manually, which in turn will stop reconnecting attempt).